cluster/addons/dns/kube2sky/Makefile

```diff
@@ -1,25 +1,66 @@
 # Makefile for the Docker image gcr.io/google_containers/kube2sky
 # MAINTAINER: Tim Hockin <thockin@google.com>
 # If you update this image please bump the tag value before pushing.
+#
+# Usage:
+#   [ARCH=amd64] [TAG=1.12] [REGISTRY=gcr.io/google_containers] [BASEIMAGE=busybox] make (build|push)
 
-.PHONY: all kube2sky container push clean test
+# Default registry, arch and tag. This can be overwritten by arguments to make
+ARCH?=amd64
+TAG?=1.12
+REGISTRY?=gcr.io/google_containers
+TEMP_DIR:=$(shell mktemp -d)
+
+# Default base images for different architectures
+# '/' in images has to be written as '\\/'
+ifeq ($(ARCH),amd64)
+	BASEIMAGE?=busybox
+endif
+ifeq ($(ARCH),arm)
+	BASEIMAGE?=hypriot\\/armhf-busybox
+endif
 
-TAG = 1.12
-PREFIX = gcr.io/google_containers
 
 all: container
 
 kube2sky: kube2sky.go
-	GOOS=linux CGO_ENABLED=0 godep go build -a -installsuffix cgo --ldflags '-w' ./kube2sky.go
+	# Only build kube2sky. This requires go
+	GOARCH=${ARCH} GOARM=6 CGO_ENABLED=0 godep go build -a -installsuffix cgo --ldflags '-w' ./kube2sky.go
+
+build:
+	# Build kube2sky in a container
+	./build-kube2sky.sh ${ARCH} ${TEMP_DIR}
+
+	# Copy the content in this dir to the temp dir
+	cp ./* ${TEMP_DIR}
+
+	# Replace BASEIMAGE with the real base image
+	cd ${TEMP_DIR} && sed -i "s/BASEIMAGE/${BASEIMAGE}/g" Dockerfile
+
+	# And build the image
+	docker build -t ${REGISTRY}/kube2sky-${ARCH}:${TAG} ${TEMP_DIR}
 
-container: kube2sky
-	docker build -t $(PREFIX)/kube2sky:$(TAG) .
+ifeq ($(ARCH),amd64)
+	# Backward compatability. TODO: deprecate this image tag
+	docker tag -f ${REGISTRY}/kube2sky-${ARCH}:${TAG} ${REGISTRY}/kube2sky:${TAG}
```

Comment:
It seems that make supports ${} but $() is more common - minor point, but I even had to go look to make sure ${} was right.